### PR TITLE
realtek: add Zyxel GS1900-8 v2

### PR DESCRIPTION
**Pull request by GoetzGoerisch**
The Zyxel GS1900-8 v2 or Rev.B1 is a newer variant of the GS1900-8, but otherwise similar to the other GS1900 switches.

Differences
------------
* Front Button labeled RESTORE
* NO Power Switch on rear
* Serial Header next to the barrel power connector
* Part Number ends 0102F

Signed-off-by: Goetz Goerisch <ggoerisch@gmail.com>

### CODE REVIEW

File: target/linux/realtek/image/rtl838x.mk [hunk 1]
```diff
@@ -313,6 +313,8 @@ define Device/zyxel_gs1900-8
   $(Device/zyxel_gs1900)
   SOC := rtl8380
   DEVICE_MODEL := GS1900-8
+  DEVICE_ALT0_VARIANT := v1
+  DEVICE_ALT1_VARIANT := v2
```
Comment (svanheule) [inline target/linux/realtek/image/rtl838x.mk:317]: You need to specifiy the primary variant with `DEVICE_VARIANT := v1`. For the first alternative (`ALT0`) you also need to specify all three variables `DEVICE_ALT0_VENDOR`, `DEVICE_ALT0_MODEL`, `DEVICE_ALT0_VARIANT`.

You can use `make menuconfig` to check if the name shows up as intended.

File: target/linux/realtek/dts-5.15/rtl8380_zyxel_gs1900-8.dts [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 / {
 	compatible = "zyxel,gs1900-8", "realtek,rtl838x-soc";
-	model = "ZyXEL GS1900-8 Switch";
+	model = "Zyxel GS1900-8v1+v2 Switch";
```
Comment (svanheule) [inline target/linux/realtek/dts-5.15/rtl8380_zyxel_gs1900-8.dts:7]: Why did you update the vendor name capitalization? It also doesn't match the one from the Makefile anymore.

I would prefer the form "GS1900-8 v1/v2" using a forward slash. "+" does not appear to be used for this purpose elsewhere.

**Review by svanheule — CHANGES_REQUESTED** [review]
Small nit about the device model name, and you'll need to update the Makefile variables.

File: target/linux/realtek/image/rtl838x.mk (hunk 1 above)
Comment (GoetzGoerisch) [reply target/linux/realtek/image/rtl838x.mk:317]: Will do. Thanks

File: target/linux/realtek/dts-5.15/rtl8380_zyxel_gs1900-8.dts (hunk 2 above)
Comment (GoetzGoerisch) [reply target/linux/realtek/dts-5.15/rtl8380_zyxel_gs1900-8.dts:7]: Will update the variant as per you suggestion.

Regarding the vendor name, you mean here: https://github.com/openwrt/openwrt/blob/8753022aeae14eaee02e0c1e59244c5b36e454db/target/linux/realtek/image/common.mk#L60

I updated, due to the fact that is no longer ZyXEL but Zyxel due to their rebranding, but overlooked the reference in the Makefile.

I was going to ask anyway is a recasing everywhere would be something useful? Then would propose this for a separate PR and keep the casing for now.
What do you prefer?

Comment (svanheule) [reply target/linux/realtek/dts-5.15/rtl8380_zyxel_gs1900-8.dts:7]: A translation of "ZyXEL" to "Zyxel" sounds like a lot of unnecessary churn to be honest... Old devices did use the old branding, so one could even consider such a global translation to be "wrong". You could open a separate PR for this, maybe others hold a different opinion.

You could use the new branding (in `DEVICE_ALT0_VENDOR`) for the new revision, but that results in unexpected sorting in `make menuconfig`

Comment (GoetzGoerisch) [reply target/linux/realtek/dts-5.15/rtl8380_zyxel_gs1900-8.dts:7]: Thanks for the suggestion, will update accordingly.

**Comment by GoetzGoerisch** [comment]
@svanheule thank you for merging.
About the casing discussion, should I ask in the IRC dev channel for opinions or just propose a PR? 

for my reference main 71ccb35

**Comment by svanheule** [comment]
The best way of judging a change in the context of an open source project would typically be to just provide a patch. In this case, your suggestion of just posting to the openwrt-devel on IRC would likely already give you some opinions.

**Comment by Neustradamus** [comment]
Good job!

**Pull request closed without merging**